toolchain/codegen/codegen.cpp

```diff
@@ -0,0 +1,65 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/codegen/codegen.h"
+
+#include <cstdio>
+
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Target/TargetMachine.h"
+#include "llvm/Target/TargetOptions.h"
+#include "llvm/TargetParser/Host.h"
+
+namespace Carbon {
+
+void PrintAssemblyFromModule(llvm::Module& module) {
+  llvm::raw_ostream& error_stream = llvm::errs();
+  // Initialize the target registry etc.
+  llvm::InitializeAllTargetInfos();
+  llvm::InitializeAllTargets();
+  llvm::InitializeAllTargetMCs();
+  llvm::InitializeAllAsmParsers();
+  llvm::InitializeAllAsmPrinters();
+
+  auto target_triple = llvm::sys::getDefaultTargetTriple();
+
+  std::string error;
+  const auto* target = llvm::TargetRegistry::lookupTarget(target_triple, error);
+
+  if (!target) {
+    error_stream << error;
+    return;
+  }
+
+  const auto* cpu = "generic";
+  const auto* features = "";
+
+  llvm::TargetOptions target_opts;
+  std::optional<llvm::Reloc::Model> reloc_model =
+      std::optional<llvm::Reloc::Model>();
```

Comment:
```suggestion
  std::optional<llvm::Reloc::Model> reloc_model;
```

No need to have the constructor call.